### DISCUSSION TITLE
关于命令的求助

### DISCUSSION THREAD

**Discussion by Zhaozhuofan** [BYJRK/DotNet-Discussions#225] [MVVM]
![image](https://github.com/BYJRK/DotNet-Discussions/assets/103978682/3be4908f-da2b-4616-b35f-a7b4f739f94c)
为什么这么写我的命令虽然绑定成功了却没有实际效果

![image](https://github.com/BYJRK/DotNet-Discussions/assets/103978682/b00cefcc-ce97-41bc-8d7b-46d474701f83)
但在构造函数中赋值的话命令就有效果

烦请各位大佬指点

**Comment by BYJRK** [comment]
你要注意，你的第一种方式，也就是 `public RelayCommand ButtonClickCommand => ...` 并不是在给这个属性初始化，而是在每次访问的时候，都在这个属性的 getter 里面重新 new 一个 `RelayCommand` 对象出来。如果你真的想初始化，你需要写成 `public RelayCommand ButtonClickCommand { get; } = ...`。但这时候你可能会发现，你如果想在这里初始化，那么会有很多限制。所以最终还是需要在构造里面进行初始化。

**Reply by Zhaozhuofan** [reply]
哦哦 原来是这样啊

**Reply by Zhaozhuofan** [reply]
谢谢UP
